### ISSUE TITLE
RFE: add an option to use hardlinks instead of symlinks

### ISSUE THREAD

**Issue by kcgthb** [gilesknap/gphotos-sync#79]
It would be great to add a command line option to use hardlinks instead of symlinks in the various linking directories (`albums/`,  `comparison/`...).

This would be optional of course, but would allow to directly copy the contents of the generated `albums/`directory structure to another host or directory, for instance, for an extra backup, or publishing. It would make the `album/` directory self-sufficient, without having to copy the `photos/` directory alongside, while still not taking additional storage space.

I can try to help contributing that feature, I'd have to look more closely at the DB structure to see what's involved.

**Comment by gilesknap** [comment]
This would be easy to implement.
- an extra option in GooglePhotosSyncMain.parser
- pass it to the GoogleAlbumsSync object
- switch on this option when calling link_file.symlink_to in create_album_content_links 

I'm happy to accept a PR if you are interested.

**Comment by kcgthb** [comment]
Here's a first shot: #81

**Comment by gilesknap** [comment]
Thanks, I have accepted (I added a system test).

Regarding the issue of switching between using --do-hardlinks and not:
I think this project has rather a large number of options now. It would be nice to be able to place a config file alongside the db  to provide some defaults. Then there would be less of an issue with switching modes (there are several other options which will not play well if used intermittently e.g. --use-flat-path)
